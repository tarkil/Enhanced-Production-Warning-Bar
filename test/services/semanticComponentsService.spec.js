import SemanticComponentsService from '../../src/js/services/semanticComponentsService'

describe('SemanticComponentsService', () => {

    beforeAll(() => {
        //Setup a stub for attachShadow. It is required by Shadow DOM
        window.HTMLElement.prototype.attachShadow = function () {
            const boundary = window.document.createElement('fake-boundary');
            this.appendChild(boundary);
            return boundary;
        };
    });

    it('Modal is rendered as a shadow dom subtree', () => {
        //When we render a warning modal
        SemanticComponentsService.renderWarningModal();
        //We expect that it is rendered as shadow dom
        expect(document.documentElement.innerHTML).toMatchSnapshot();
    });
});