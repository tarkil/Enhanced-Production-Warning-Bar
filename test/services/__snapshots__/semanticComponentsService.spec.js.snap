// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemanticComponentsService Modal is rendered as a shadow dom subtree 1`] = `
"<head><style>
                        @font-face {
                            font-family: 'Icons';
                            src: url(\\"build/js/icons.eot\\");
                            src: url(\\"build/js/icons.eot?#iefix\\") format('embedded-opentype'), url(\\"build/js/icons.ttf\\") format('truetype'), url(\\"build/js/icons.svg#icons\\") format('svg');
                            font-style: normal;
                            font-weight: normal;
                            font-variant: normal;
                            text-decoration: inherit;
                            text-transform: none;
                          }
                        </style></head><body><div id=\\"vxz01wnfm11cwlvxi529\\"><div class=\\"resolving\\"><fake-boundary><div class=\\"ui active transition visible dimmer\\"><div class=\\"content\\"><div class=\\"center\\"><h2 class=\\"ui icon inverted header\\"><i aria-hidden=\\"true\\" class=\\"industry icon\\"></i>warningModal_title<div class=\\"sub header\\">warningModal_description</div></h2><div class=\\"ui container\\"><button class=\\"ui red basic inverted button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"remove icon\\"></i>warningModal_exit</button><button class=\\"ui green icon inverted right labeled button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"chevron right icon\\"></i>warningModal_continue</button></div></div></div></div></fake-boundary></div></div></body>"
`;
